src/test/regression.c

```diff
@@ -4945,6 +4945,29 @@ static void test_remove_tzid_from_due(void)
     icalcomponent_free(c);
 }
 
+void test_icaltime_as_timet(void) {
+
+    ok("icaltime_from_string translates 19020101T000000Z to -2145916800", icaltime_as_timet(icaltime_from_string("19020101T000000Z")) == -2145916800);
+    ok("icaltime_from_string translates 19290519T000000Z to -1281916800", icaltime_as_timet(icaltime_from_string("19290519T000000Z")) == -1281916800);
+    ok("icaltime_from_string translates 19561004T000000Z to -417916800", icaltime_as_timet(icaltime_from_string("19561004T000000Z")) == -417916800);
+    ok("icaltime_from_string translates 19700101T000000Z to 0", icaltime_as_timet(icaltime_from_string("19700101T000000Z")) == 0);
+    ok("icaltime_from_string translates 19700301T235959Z to 5183999", icaltime_as_timet(icaltime_from_string("19700301T235959Z")) == 5183999);
+    ok("icaltime_from_string translates 19970717T235959Z to 869183999", icaltime_as_timet(icaltime_from_string("19970717T235959Z")) == 869183999);
+    ok("icaltime_from_string translates 20241202T235959Z to 1733183999", icaltime_as_timet(icaltime_from_string("20241202T235959Z")) == 1733183999);
+    ok("icaltime_from_string translates 20371231T235959Z to 2145916799", icaltime_as_timet(icaltime_from_string("20371231T235959Z")) == 2145916799);
+#if (SIZEOF_ICALTIME_T > 4)
```

Comment:
the 3.0 branch doesn't have a macro for SIZEOF_ICALTIME_T 
I guess you'll need to add:
 * `check_type_size(${ICAL_ICALTIME_T_TYPE} SIZEOF_ICALTIME_T)` to ConfigureChecks.cmake
 * `#cmakedefine SIZEOF_ICALTIME_T ${SIZEOF_ICALTIME_T}` to config.h.cmake

Reply:
Thanks! We don't have icaltime_t in 3.0. We're still using time_t, so we can simply check SIZEOF_TIME_T. Fixed it.